eth/api.go

```diff
@@ -671,3 +671,30 @@ func (api *ScrollAPI) GetLatestRelayedQueueIndex(ctx context.Context) (queueInde
 	lastIncluded := *queueIndex - 1
 	return &lastIncluded, nil
 }
+
+// rpcMarshalBlock uses the generalized output filler, then adds the total difficulty field, which requires
+// a `ScrollAPI`.
+func (api *ScrollAPI) rpcMarshalBlock(ctx context.Context, b *types.Block, inclTx bool, fullTx bool) (map[string]interface{}, error) {
+	fields, err := ethapi.RPCMarshalBlock(b, inclTx, fullTx, api.eth.APIBackend.ChainConfig())
+	if err != nil {
+		return nil, err
+	}
+	if inclTx {
+		fields["totalDifficulty"] = (*hexutil.Big)(api.eth.APIBackend.GetTd(ctx, b.Hash()))
+	}
```

Comment:
Why do we need to add this here?

Reply:
It comes from here https://github.com/scroll-tech/go-ethereum/blob/38a6a1e964e56d1cd8abb0f1bd4e3591385645e9/internal/ethapi/api.go#L1307

Reply:
If we only pass `true` we might as well remove the `if`.

Reply:
Ok, I see what you mean